### ISSUE TITLE
cost_j_h = (beta_v + 0.5*tf.log(var_j)) ?

### ISSUE THREAD

**Issue by itxiud2015** [IBM/matrix-capsules-with-em-routing#10]
Hi Ashley,

For 'def m_step()' in em_routing.py, I can see you code 'cost_j_h = (beta_v + 0.5*tf.log(var_j)) * rr_prime_sum * layer_norm_factor' prior 'cost_j = tf.reduce_sum(cost_j_h, axis=-1, keepdims=True, name="cost_j")' .
My question is whether this lead to beta_v been multiplied by 'h' times? because 'beta_v + 0.5*tf.log(var_j)' will broadcast beta_v over all the h elements along the last dimension. 
According formula (2) in 'M ATRIX CAPSULES WITH EM
ROUTING' paper, it should be put something like (beta_v + Sum of cost_j_h) instead of Sum of (beta_v + cost_j_h), how do you think? maybe i am wrong.

Kindly regards
Jeff

**Comment by ashleygritzman** [comment]
Hi Jeff,

Thanks for the question, when I look again at the paper, specifically the M-STEP (ln 5) in Procedure 1, the way it is written does suggest that it should be broadcast across h dimensions (h=16 in this case), what do you think?

I don't recon it makes too much difference either way, because beta_v is learned during training, so the network could just learn a value for beta_v that is 16 times smaller, and then we could get to the same place.

**Comment by itxiud2015** [comment]
Hi Ashley,

I just checked M-STEP (ln 5) in Procedure 1 in the paper also, yes, you are right.
And I agree with your argument.

**Comment by ashleygritzman** [comment]
👍
